libraries/stdlib/jvm/src/kotlin/coroutines/jvm/internal/boxing.kt

```diff
@@ -25,24 +25,24 @@ internal fun boxByte(primitive: Byte): java.lang.Byte = java.lang.Byte.valueOf(p
 
 @SinceKotlin("1.3")
 @PublishedApi
-internal fun boxShort(primitive: Short): java.lang.Short = java.lang.Short(primitive)
+internal fun boxShort(primitive: Short): java.lang.Short = java.lang.Short.valueOf(primitive) as java.lang.Short
```

Comment:
These constructor calls are here for the reason: they enable sophisticated HotSpot optimizations in coroutines heavy code. In this case, HotSpot scalarizes primitive wrappers on fast-path, while on slow-path (i.e. suspension) it keeps the wrappers. Using `valueOf` breaks the optimization, slowing down coroutines code.
Besides, we do know about the warning, it has been there since JDK 9 and we deliberately use wrapper constructors for the reason above.

Reply:
Yep, it's also explained in a comment at the top of this file.